apps/console/src/features/console-settings/components/console-administrators/add-existing-user-wizard/add-existing-user-wizard.tsx

```diff
@@ -0,0 +1,306 @@
+ * under the License.
+ */
+
+import { AutocompleteRenderGetTagProps } from "@oxygen-ui/react/Autocomplete";
+import Chip from "@oxygen-ui/react/Chip";
+import Typography from "@oxygen-ui/react/Typography";
+import { AlertLevels, IdentifiableComponentInterface, RolesInterface } from "@wso2is/core/models";
+import { AutocompleteFieldAdapter, FinalForm, FinalFormField, FormRenderProps } from "@wso2is/form";
+import { Heading, Hint, LinkButton, PrimaryButton, useWizardAlert } from "@wso2is/react-components";
+import { AxiosError } from "axios";
+import isEmpty from "lodash-es/isEmpty";
+import React, { FunctionComponent, ReactElement, ReactNode, useMemo } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { Dispatch } from "redux";
+import { Grid, Modal, ModalProps } from "semantic-ui-react";
+import { UsersConstants } from "../../../../../extensions/components/users/constants/users";
+import { UserBasicInterface } from "../../../../core/models/users";
+import { ConsoleAdministratorOnboardingConstants } from "../../../constants/console-administrator-onboarding-constants";
+import useBulkAssignAdministratorRoles from "../../../hooks/use-bulk-assign-user-roles";
+import useConsoleRoles from "../../../hooks/use-console-roles";
+import useProspectiveAdministrators from "../../../hooks/use-prospective-administrators";
+import "./add-existing-user-wizard.scss";
+
+/**
+ * Props interface of {@link AddExistingUserWizard}
+ */
+export interface AddExistingUserWizardPropsInterface extends IdentifiableComponentInterface, ModalProps {
+    onSuccess?: () => void;
+}
+
+interface AddExistingUserWizardFormValuesInterface {
+    username: {
+        key: string;
+        label: ReactNode;
+        user: UserBasicInterface;
+    };
+    roles: {
+        key: string;
+        label: ReactNode;
+        role: RolesInterface;
+    }[];
+}
+
+interface AddExistingUserWizardFormErrorsInterface {
+    username: string;
+    roles: string;
+}
+
+/**
+ * Component to handle add existing user wizard.
+ *
+ * @param props - Props injected to the component.
+ * @returns Add existing user wizard component.
+ */
+const AddExistingUserWizard: FunctionComponent<AddExistingUserWizardPropsInterface> = (
+    props: AddExistingUserWizardPropsInterface
+): ReactElement => {
+    const {
+        onClose,
+        onSuccess,
+        ["data-componentid"]: componentId,
+        ...rest
+    } = props;
+
+    const { t } = useTranslation();
+
+    const dispatch: Dispatch = useDispatch();
+
+    const [ alert, setAlert, alertComponent ] = useWizardAlert();
+
+    const { consoleRoles } = useConsoleRoles(null, null);
+
+    const { assignAdministratorRoles } = useBulkAssignAdministratorRoles();
+
+    const { prospectiveAdministrators } = useProspectiveAdministrators(
+        null,
+        null,
+        null,
+        null,
+        null,
+        UsersConstants.GROUPS_ATTRIBUTE,
+        true
+    );
```

Comment:
If useProspectiveAdministrators can work without any params passed, we can make it's params optional.

Also, instead of passing multiple params, we can define a single object param with these values, IMO.